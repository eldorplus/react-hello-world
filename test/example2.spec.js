import React from 'react';
import Example2 from '../src/example2';
import renderer from 'react-test-renderer';

test('Example2 toggles Message after 3 clicks', () => {
  const component = renderer.create(
    <Example2 numClicks={3} />
  );
  let tree = component.toJSON();

  tree.children[1].props.onClick();
  tree.children[1].props.onClick();
  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].props.onClick();
  tree.children[1].props.onClick();
  tree.children[1].props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
