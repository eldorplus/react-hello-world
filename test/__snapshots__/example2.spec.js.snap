exports[`test Example2 toggles Message after 3 clicks 1`] = `
<div>
  <h2>
    Example 2 - 
    <small>
      click 
      3
       times to toggle
    </small>
  </h2>
  <input
    onClick={[Function]}
    type="submit"
    value="Toggle message" />
</div>
`;

exports[`test Example2 toggles Message after 3 clicks 2`] = `
<div>
  <h2>
    Example 2 - 
    <small>
      click 
      3
       times to toggle
    </small>
  </h2>
  <input
    onClick={[Function]}
    type="submit"
    value="Toggle message" />
</div>
`;
